lib/loggable.ex

```diff
@@ -62,9 +97,15 @@ defimpl StathamLogger.Loggable, for: Map do
   defp sanitize_map_key(key) when is_binary(key) or is_atom(key) or is_number(key), do: key
   defp sanitize_map_key(key), do: inspect(key)
 
-  defp should_be_filtered(field, sensitive_keys) do
-    Enum.any?(sensitive_keys, fn key -> to_string(key) == to_string(field) end)
+  defp should_be_filtered(field, {:keep, keys = [_h | _t]}) do
+    Enum.any?(keys, fn key -> to_string(key) != to_string(field) end)
   end
+
+  defp should_be_filtered(field, {:discard, keys = [_h | _t]}) do
+    Enum.any?(keys, fn key -> to_string(key) == to_string(field) end)
+  end
```

Comment:
Alternative implementation for this function:
```elixir
  defp should_be_filtered(field, {operation, keys}) when is_list(keys) do
  comparator =
    case operation do
      :keep -> &==/2
      :discard -> &!=/2
    end

  field = to_string(field)
  Enum.any?(keys, fn key -> comparator.(to_string(key), field) end)
end
```

Reply:
My code had a bug - I had to use `Enum.all?` for `keep` and update test.